### ISSUE TITLE
Some Configuration settings not getting respected in v11.0.0

### ISSUE THREAD

**Issue by nettum** [flameshot-org/flameshot#2271]
### Flameshot Version

Flameshot v11.0.0 ()
Compiled with Qt 5.15.3
linux: 4.15.0-166-generic
ubuntu-core: 20

### Installation Type

Snap from Snapcraft

### Operating System type and version

Ubuntu 18.04.6 LTS

### Description

Seems like at least the settings "preferred save file extension" and "Filename Editor" is not getting used when trying  to save a file.
I would expect that the filename and the save extension would be set to what I have in the configuration. See attached screenshots for settings and save dialogue.

(For the record I've also got the "The configuration contains an error. Open configuration to resolve" multiple times without there being an obvious error after update. It's gone now though)

### Steps to reproduce

1. Edit "preferred save file extension" and/or "Filename Editor" in configuration
2. Use the "Take Screenshot" launcher and press the "Save screenshot to a file"-button after choosing the area you want to screenshot
3. Check that the save dialog does not use the filename or the extension you added in the config.

### Screenshots or screen recordings

![image](https://user-images.githubusercontent.com/798156/149746529-dae5160d-7282-4bb6-ad6e-59e483fdfa7d.png)
![image](https://user-images.githubusercontent.com/798156/149746601-9b6b5aad-ffe7-406f-a82d-15f87951e83e.png)
![image](https://user-images.githubusercontent.com/798156/149746937-910b1410-8595-44bc-b237-b807676fc3e5.png)

### System Information

![image](https://user-images.githubusercontent.com/798156/149747904-4d88bd71-beb1-4fc4-9e43-8148bb67f947.png)

**Comment by nettum** [comment]
And here is my `flameshot.ini`:

```
▶ cat snap/flameshot/current/.config/flameshot/flameshot.ini 
[General]
contrastOpacity=188
disabledTrayIcon=false
drawColor=#ff1818
drawFontSize=8
drawThickness=2
filenamePattern=%F%T
ignoreUpdateToVersion=0.10.2
saveAsFileExtension=png
savePath=/home/<something>/<something>
savePathFixed=false
showStartupLaunchMessage=true
startupLaunch=true

[Shortcuts]
TYPE_ARROW=A
TYPE_CIRCLE=C
TYPE_CIRCLECOUNT=
TYPE_COMMIT_CURRENT_TOOL=Ctrl+Return
TYPE_COPY=Ctrl+C
TYPE_DELETE_CURRENT_TOOL=Del
TYPE_DRAWER=D
TYPE_EXIT=Ctrl+Q
TYPE_IMAGEUPLOADER=Return
TYPE_MARKER=M
TYPE_MOVESELECTION=Ctrl+M
TYPE_MOVE_DOWN=Down
TYPE_MOVE_LEFT=Left
TYPE_MOVE_RIGHT=Right
TYPE_MOVE_UP=Up
TYPE_OPEN_APP=Ctrl+O
TYPE_PENCIL=P
TYPE_PIN=
TYPE_PIXELATE=B
TYPE_RECTANGLE=R
TYPE_REDO=Ctrl+Shift+Z
TYPE_RESIZE_DOWN=Shift+Down
TYPE_RESIZE_LEFT=Shift+Left
TYPE_RESIZE_RIGHT=Shift+Right
TYPE_RESIZE_UP=Shift+Up
TYPE_SAVE=Ctrl+S
TYPE_SELECTION=S
TYPE_SELECTIONINDICATOR=
TYPE_SELECT_ALL=Ctrl+A
TYPE_TEXT=T
TYPE_TOGGLE_PANEL=Space
TYPE_UNDO=Ctrl+Z

```

**Comment by borgmanJeremy** [comment]
I am unable to replicate this in an Ubuntu VM with the snap install. Can you try deleting the whole config file and see if that resolves it?

**Comment by nettum** [comment]
Sure! Just tried to delete it and then change the preferred extension. The config file now just looks like this:
```
▶ cat flameshot.ini                         
[General]
contrastOpacity=181
saveAsFileExtension=png
```
But it did not have any effect. I will try on another Ubuntu version to check if it works as expected there.

**Comment by borgmanJeremy** [comment]
Okay thanks, I tested on 21.10 but I can stand up a vm to match your version later.

**Comment by nettum** [comment]
I just tested this on 20.04 and it remembers the PNG extension when opening the save dialogue, but not the filename pattern.

That being said, I may have misunderstood how the settings should work.
If I set `savePathFixed=true` and takes a screenshot it will save the file according to `savePath`, `filenamePattern` and `saveAsFileExtension` correctly :+1: 
But if I set `savePathFixed=false` I was thinking that that suggested filename was going to be e.g. `2022-01-17-18-32.png` (not getting this on 18.04 or 20.04) and that it would be set to "PNG" and not "BMP" in the suggested extension (working on 20.04, not 18.04).
See screenshot.

If this is working as expected you can close this, if not I can  try a clean 18.04 VM to see if the problem exists there.
![image](https://user-images.githubusercontent.com/798156/149746937-910b1410-8595-44bc-b237-b807676fc3e5.png)

**Comment by borgmanJeremy** [comment]
Yeah that is not the behavior I see: 
![image](https://user-images.githubusercontent.com/46930769/149836514-f23b3477-e76d-4ba0-a5d7-21c70e8b4773.png)

**Comment by nettum** [comment]
All right, good to know. That is the same behavior that I got at 20.04, so probably something with the system I'm running 19.04 on. I can test with a clean VM, but this is just a minor thingy.

But how about the `filenamePattern` if you had that config set? Should that be added as the suggested filename (along with the extension) in the "Name"-input from your screenshot? 
I _may_ remember wrong, but I believe that was the behavior  in earlier versions?

**Comment by mmahmoudian** [comment]
> I _may_ remember wrong, but I believe that was the behavior in earlier versions?

As far as I can see, it still is:
https://user-images.githubusercontent.com/390889/149918640-b1208b7d-dc47-45e6-a700-d438da58d568.mp4

**Comment by nettum** [comment]
Awesome, thanks! I will test on different clean Ubuntu VMs then to see if I can replicate the bug(s) there!

**Comment by nettum** [comment]
Getting the same behavior with missing filename and extension on a clean Ubuntu 18.04.
Will test on 20.04 and 21.10 a bit later.



https://user-images.githubusercontent.com/798156/149946061-bc02779e-74de-423f-9594-cc8c6d6982e0.mp4

**Comment by nettum** [comment]
Clean Ubuntu 20.04, file extension are working, but not filename:

https://user-images.githubusercontent.com/798156/149955087-ac2a2169-046b-40eb-b1f0-e090d713764b.mp4

**Comment by mmahmoudian** [comment]
> but I'm getting some weird behavior(?)

Nope, this is because of the change Gnome people did in Gnome 41 Wayland. Checkout #1910 they have limited all applications other than their own apps from using the API, therefore users should go through multiple hoops. Anyways, we will no do anything for that as it is an upstream issue and you should take your complaints to gnome devs (all necessary links in #1910)

As for your other two videos, it is certainly unexpected and unintended. Needs some investigation.

**Comment by nettum** [comment]
Got it, thanks for clarifying! Will check that issue. Let me know if I should do some more testing. 
For the record, according to neofetch a clean Ubuntu 21.10 are running gnome 40.5.
![image](https://user-images.githubusercontent.com/798156/150028791-778dbaf6-5f82-4782-a48f-d94deb86fb78.png)

**Comment by borgmanJeremy** [comment]
Thanks for all the details. This is a problem in the snap only, when I build from source on the same platform the problem is gone.

Ill troubleshoot to see if the issue is related to the gnome libraries bundled in the snap, or a problem _in flameshot_ running in the snap.
